src/bindings/python/flux/job/Jobspec.py

```diff
@@ -822,6 +822,57 @@ def _v1_validate(resources, tasks, kwargs):
             if task["count"].keys().isdisjoint({"per_slot", "total"}):
                 raise ValueError("count per_slot or total must be set")
 
+    def _set_extra_options(
+        self,
+        duration=None,
+        environment=None,
```

Comment:
I really do wonder if we should consider setting `environment=dict(os.environ)` by default. There are good reasons not to, but since it's something we _consistently_ end up explaining to users, I wanted to start the discussion.

Reply:
Or maybe a convenience option to the `.from_command` method, like `add_env=True`

Reply:
This is a good suggestion. We should open an issue on having the default copy the current environment.  I'm not sure of the good reasons not to except that we haven't before.

Note that using `environment=os.environ` now would be the same as `add_env=True`.

Reply:
Thanks @grondo!